packages/account/src/Components/poi/idv-document-submit/idv-document-submit.tsx

```diff
@@ -18,220 +17,233 @@ import BackButtonIcon from 'Assets/ic-poi-back-btn.svg';
+
+            if (values.first_name) {
+                errors.first_name = validateName(values.first_name);
+            }
+            if (values.last_name) {
+                errors.last_name = validateName(values.last_name);
+            }
+
+            return removeEmptyPropertiesFromObject(errors);
         };
 
-        if (submit_data.document_type === IDV_NOT_APPLICABLE_OPTION.id) {
-            return;
-        }
-        WS.send(submit_data).then(response => {
-            setSubmitting(false);
-            if (response.error) {
-                setErrors({ error_message: response.error.message });
+        const submitHandler = async (values, { setSubmitting, setErrors }) => {
+            setSubmitting(true);
+
+            const request = makeSettingsRequest(values, changeable_fields);
+
+            const data = await WS.setSettings(request);
+
+            if (data.error) {
+                setErrors({ error_message: data.error.message });
+                setSubmitting(false);
                 return;
             }
-            handleViewComplete();
-        });
-    };
-
-    return (
-        <Formik initialValues={{ ...initial_values }} validate={validateFields} onSubmit={submitHandler}>
-            {({
-                dirty,
-                errors,
-                handleBlur,
-                handleChange,
-                handleSubmit,
-                isSubmitting,
-                isValid,
-                setFieldValue,
-                setFieldTouched,
-                touched,
-                values,
-            }) => (
-                <div className='proof-of-identity__container proof-of-identity__container--reset'>
-                    <section className='form-body'>
-                        <FormSubHeader title={localize('Identity verification')} />
-                        <IDVForm
-                            errors={errors}
-                            touched={touched}
-                            values={values}
-                            handleChange={handleChange}
-                            handleBlur={handleBlur}
-                            setFieldValue={setFieldValue}
-                            hide_hint={false}
-                            selected_country={selected_country}
-                            is_from_external={is_from_external}
-                            class_name='idv-layout'
-                        />
-
-                        <FormSubHeader title={localize('Details')} />
-                        <div
-                            className={classNames({
-                                'account-form__poi-confirm-example_container': !shouldHideHelperImage(
-                                    values?.document_type?.id
-                                ),
-                            })}
-                        >
-                            <PersonalDetailsForm
+            const get_settings = WS.authorized.storage.getSettings();
```

Comment:
should we 'await' for fulfilled promise here?

Reply:
Not sure about this as this was already there

Reply:
I think we need, could you please check it